lib/galaxy/webapps/galaxy/api/__init__.py

```diff
@@ -602,3 +606,26 @@ def search_query_param(model_name: str, tags: list, free_text_fields: list) -> O
         title="Search query.",
         description=description,
     )
+
+
+# TODO: mypy complains that Depends(get_body) is returned B is expected.
+# def depend_on_either_json_or_form_data(model: Type[B]) -> B:
+def depend_on_either_json_or_form_data(model: Type[B]):
+    async def get_body(request: Request):
+        content_type = request.headers.get("Content-Type")
+        if content_type is None:
+            raise HTTPException(status_code=400, detail="No Content-Type provided!")
```

Comment:
Same here, probably better to use `MessageException`. I know this function comes from the ToolShed code but I don't see why it should use `HTTPException` instead of galaxy exceptions as the rest of the code there, even if it is just for consistency :thinking: